lua/lspconfig/server_configurations/julials.lua

```diff
@@ -1,5 +1,59 @@
 local util = require 'lspconfig.util'
 
+local function activate_env(path)
+  local bufnr = vim.api.nvim_get_current_buf()
+  local julials_clients = (vim.lsp.get_clients or vim.lsp.get_active_clients) { bufnr = bufnr, name = 'julials' }
+  if #julials_clients == 0 then
+    vim.notify(
+      'method julia/activateenvironment is not supported by any servers active on the current buffer',
+      vim.log.levels.WARN
+    )
+    return
+  end
+  local julia_project_files = { 'Project.toml', 'JuliaProject.toml' }
+  local function _activate_env(environment)
+    if environment then
+      for _, julials_client in ipairs(julials_clients) do
+        julials_client.notify('julia/activateenvironment', { envPath = environment })
+      end
+      vim.notify('Julia environment activated: \n`' .. environment .. '`', vim.log.levels.INFO)
+    end
+  end
+  if path then
+    path = vim.fs.normalize(path)
+    local found_env = false
+    for _, project_file in ipairs(julia_project_files) do
+      local file = (vim.uv or vim.loop).fs_stat(vim.fs.joinpath(path, project_file))
+      if file and file.type then
+        found_env = true
+        break
+      end
+    end
```

Comment:
can this loop be replaced by `vim.fs.find` ? Nvim versions before 0.10 are not supported (especially for commands)

Reply:
I don't think so because `vim.fs.find` doesn't have a maximum depth. It has a `stop` parameter which stops when a given directory is reached but this really is only applicable doing an upwards search and there is no way to limit the depth it searches. Here we need to find project files just inside the directory that is passed in and not looking for them infinitely recursing down.